src/main/java/seedu/address/logic/commands/CategorizeTagCommand.java

```diff
@@ -26,33 +28,36 @@ public class CategorizeTagCommand extends Command {
     public static final String MESSAGE_TAG_NOT_EXIST = "Tag not found: %1$s";
     public static final String MESSAGE_INVALID_CATEGORY = "Invalid category: %1$s";
     public static final String MESSAGE_DUPLICATE_CATEGORY = "Current category of %s is already %s";
-
-    private final Tag targetTag;
+    private final List<Tag> targetTags;
     private final TagCategory updatedCategory;
 
     /**
      * Constructs a command to categorize a tag.
      */
-    public CategorizeTagCommand(Tag targetTag, TagCategory updatedCategory) {
-        requireNonNull(targetTag);
+    public CategorizeTagCommand(List<Tag> targetTags, TagCategory updatedCategory) {
+        requireNonNull(targetTags);
         requireNonNull(updatedCategory);
-        this.targetTag = targetTag;
+        this.targetTags = targetTags;
         this.updatedCategory = updatedCategory;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         assert model != null;
-        if (!model.containsTag(targetTag)) {
-            throw new CommandException(String.format(MESSAGE_TAG_NOT_EXIST, targetTag));
-        }
-        TagCategory existingCat = model.getTagCategory(targetTag);
-        if (isDuplicateCategory(existingCat)) {
-            throw new CommandException(String.format(MESSAGE_DUPLICATE_CATEGORY, targetTag, updatedCategory));
+        for (Tag targetTag : targetTags) {
+            if (!model.containsTag(targetTag)) {
+                throw new CommandException(String.format(MESSAGE_TAG_NOT_EXIST, targetTag));
+            }
+
+            TagCategory existingCat = model.getTagCategory(targetTag);
+            if (isDuplicateCategory(existingCat)) {
+                throw new CommandException(String.format(MESSAGE_DUPLICATE_CATEGORY, targetTag, updatedCategory));
+            }
+            model.setTagsCategory(targetTag, updatedCategory);
+            model.refreshCampusConnect();
         }
-        model.setTagsCategory(targetTag, updatedCategory);
-        model.refreshCampusConnect();
-        return new CommandResult(String.format(MESSAGE_CAT_TAG_SUCCESS, targetTag));
+
+        return new CommandResult(String.format(MESSAGE_CAT_TAG_SUCCESS, targetTags));
```

Comment:
Sorry, last comment. Should we make the command successful only when all the tags are valid? If not the user might not get feedback that one tag succeeded but the other did not.
Maybe we should loop through the tags and check duplicate + validity, before setting the categories and refreshing.

Reply:
Hmm but then we would need to change the formatting of the display, add in a new message for that scenario. Would it be easier to just disallow the whole imput.

Reply:
> Would it be easier to just disallow the whole input.

Yeah thats what I mean, we should verify all the tags first before doing the model logic outside the loop.
No worries, we merge first, I will add on in another PR.